kudu-handler/src/java/org/apache/hadoop/hive/kudu/KuduSerDe.java

```diff
@@ -0,0 +1,263 @@
+  }
+
+  private static ObjectInspector createObjectInspector(Schema schema) throws SerDeException {
+    Preconditions.checkNotNull(schema);
+    List<String> fieldNames = new ArrayList<>();
+    List<ObjectInspector> fieldInspectors = new ArrayList<>();
+    List<String> fieldComments = new ArrayList<>();
+    for (int i = 0; i < schema.getColumnCount(); i++) {
+      ColumnSchema col = schema.getColumnByIndex(i);
+      PrimitiveTypeInfo typeInfo = toHiveType(col.getType(), col.getTypeAttributes());
+      fieldNames.add(col.getName());
+      fieldInspectors.add(
+          PrimitiveObjectInspectorFactory.getPrimitiveWritableObjectInspector(typeInfo));
+      fieldComments.add(col.getComment());
+    }
+    return ObjectInspectorFactory.getStandardStructObjectInspector(fieldNames, fieldInspectors,
+        fieldComments);
+  }
+
+  public Schema getSchema() {
+    return schema;
+  }
+
+  @Override
+  public ObjectInspector getObjectInspector() {
+    return objectInspector;
+  }
+
+  @Override
+  public Class<? extends Writable> getSerializedClass() {
+    return KuduWritable.class;
+  }
+
+  /**
+   * Serialize an object by navigating inside the Object with the ObjectInspector.
+   */
+  @Override
+  public KuduWritable serialize(Object obj, ObjectInspector objectInspector) throws SerDeException {
+    Preconditions.checkArgument(objectInspector.getCategory() == Category.STRUCT);
+
+    StructObjectInspector soi = (StructObjectInspector) objectInspector;
+    List<Object> writableObj = soi.getStructFieldsDataAsList(obj);
+    List<? extends StructField> fields = soi.getAllStructFieldRefs();
+
+    PartialRow row = schema.newPartialRow();
+    for (int i = 0; i < schema.getColumnCount(); i++) {
+      StructField field = fields.get(i);
+      Object value = writableObj.get(i);
+
+      if (value == null) {
+        row.setNull(i);
+      } else {
+        Type type = schema.getColumnByIndex(i).getType();
+        ObjectInspector inspector = field.getFieldObjectInspector();
+        switch (type) {
+        case BOOL:
+          boolean boolVal = ((BooleanObjectInspector) inspector).get(value);
+          row.addBoolean(i, boolVal);
+          break;
+        case INT8:
+          byte byteVal = ((ByteObjectInspector) inspector).get(value);
+          row.addByte(i, byteVal);
+          break;
+        case INT16:
+          short shortVal = ((ShortObjectInspector) inspector).get(value);
+          row.addShort(i, shortVal);
+          break;
+        case INT32:
+          int intVal = ((IntObjectInspector) inspector).get(value);
+          row.addInt(i, intVal);
+          break;
+        case INT64:
+          long longVal = ((LongObjectInspector) inspector).get(value);
+          row.addLong(i, longVal);
+          break;
+        case UNIXTIME_MICROS:
+          java.sql.Timestamp timestampVal = ((TimestampObjectInspector) inspector)
+              .getPrimitiveJavaObject(value).toSqlTimestamp();
```

Comment:
Same as above, please verify that this method is returning the timestamp object that you expect.